### PR TITLE
[DLSP21] week 11-1

### PR DESCRIPTION
**Pull request by calpeyser**

### CODE REVIEW

File: docs/en/week11/11-1.md.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+---
```
Comment (Atcold) [inline docs/en/week11/11-1.md.md:1]: File name is wrong (there are 2 mds)!

File: docs/en/week11/11-1.md.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+lang-ref: ch.11-1
+lecturer: Awni Hannun
+title: Speech Recognition and Graph Transformer Network
+authors: Cal Peyser, Kevin Chang
+date: 14 April 2021
```
Comment (Atcold) [inline docs/en/week11/11-1.md.md:6]:
```suggestion
date: 14 Apr 2021
```

File: docs/en/week11/11-1.md.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+Given some input speech utterance $X$, which consists of $T$ frames of audio. We desire to produce a transcription $Y$ and we'll think of our transcription as consisting of the letters of a sentence, so $y_1$ is the first letter $y_U$ is the last letter.
+
+$$
+X=[x_1,...,x_T],\ Y=[y_1,...,y_U]
+$$
+
+Compute conditional probability(the score) to evaluate transcription, we want to maximize the probability.
+
+$$\log{P(Y \vert X;\theta)}$$
```
Comment (Atcold) [inline docs/en/week11/11-1.md.md:42]: The conditional bar is `\mid`. Here and elsewhere.

File: docs/en/week11/11-1.md.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    
+**Reminder**: use actual-softsoftmax to sum log probabilities.
+
+We want $\log{(P_1+P_2)}$ from $\log{P_1}$ and $\log{P_2}$
+
+$$\begin{align}
+\text{actual-softmax}(\log{P_1}, \log{P_2}) 
+&= \log{P_1}+\log{P_2} \\
+&= \log{(e^{\log{P1}}+e^{\log{P2}})}
+\end{align}$$
```
Comment (Atcold) [inline docs/en/week11/11-1.md.md:80]: Please, test locally.
image

File: docs/en/week11/11-1.md.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+    
+### Alignment graph
+
+Alignment graph is a way to encode the set of possible alignments to an arbitrary length input.
+    
+<center>
+<img src="{{site.baseurl}}/images/week11/11-1/figure1.png" style="zoom: 40%; background-color:#DCDCDC;"/><br>
```
Comment (Atcold) [inline docs/en/week11/11-1.md.md:87]: image

File: docs/en/week11/11-1.md.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+$$
+
+This is the naive approch to compute $\alpha_3^a$.
+
+Using this forward variable, we seek to model the probablity distribution $P(Y | X) = \sum_{a \in A} P(a)$, where $A$ is the set of all possible alignments from $Y$ to $X$.  This decomposes as 
+
+$$P(Y|X) = \sum_{a \in A}  \prod_{t=1}^T P(a_t | X)$$
+
+where $P(a_t | X)$ are the output logits of a system such as an RNN. That is, to compute the likelihood of the transcript $Y$ we must marginalize over an intractably large number of alignments.  We may do this with a recursive decomposition of the forward variable.  The below presentation is inspired by https://distill.pub/2017/ctc/, which is an excellent introduction to the algorithm.
+
```
Comment (Atcold) [inline docs/en/week11/11-1.md.md:130]: image
Use `\mid`!

File: docs/en/week11/11-1.md.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+Now, suppose $y_i = y_{i+1}$, so that $Z$ contains a subsequence $y_i, \epsilon, y_{i+1}$, and suppose $y_{i+1}$ occurs at position $s$ in $Z$.  Then the alignment for $\alpha_{s}^t$ can be arrived at by one of two ways: either the prediction at time $t-1$ can be $y_{i+1}$ (in which case the repetition is collapesed) or else the prediction at time $t-1$ can be epsilon.  So, we may decompose:
+
+$$\alpha_s^t = (\alpha_{s, t-1} + \alpha_{s-1, t-1}) P(z_s|X)$$
+
+where the elements of the sum represent the two possible prefixes to the alignment.  If, on the other hand, we have $y_i \ne y_{i+1}$ then there is the additional third possiblity that the prediction at time $t-1$ is equal to $y_i$.  So, we have the decomposition
+
+$$\alpha_s^t = (\alpha_{s, t-1} + \alpha_{s-1, t-1} + \alpha{s-2, t-1}) P(z_s|X)$$
+
+By computing $\alpha_{|Z|}^{T}$, we may effectively marginalize over all possible alignments between the transcript $Y$ and the audio $X$, allowing efficient training and inference.  This is called Connectionist Temporal Classification, or CTC.
```
Comment (Atcold) [inline docs/en/week11/11-1.md.md:141]: image

File: docs/en/week11/11-2.md [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+title: Speech Recognition and Graph Transformer Network (Part 2)
+authors: Gyanesh Gupta, Abed Qaddoumi
+date: 14th April 2021
+---
+
+## Inference Time
+
+The inference of a transcription from a given audio signal can be formulated using 2 distributions:
+
+* The acoustic model (audio to transcription), represented as $P(Y|X)$
```
Comment (Atcold) [inline docs/en/week11/11-2.md:13]: `\mid`!

File: docs/en/week11/11-2.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+We use the additional term to ensure the transcription is consistent with the rules of the language. Else we may get grammatically wrong transcription.
+
+
+## Beam Search
+
+While returning an output sequence, we can follow the greedy approach, where we take the maximum value of $$P(y_{t} | P(y_{t-1}...y_{1}))$$
+
+However, we can end up missing out a good sequence, which may not have the maximum value of $P(y_{t} | ...)$, as illustrated by the example below.
+
+![Shortest Path failing in the Greedy Approach](https://i.imgur.com/DX8j0Sv.png)
```
Comment (Atcold) [inline docs/en/week11/11-2.md:29]: ???

File: docs/en/week11/11-2.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+To remedy this, we employ beam search. Essentially, we consider maximum $k$ tokens in terms of probability at each step $t$ of the sequence. For each of these n-grams, we proceed further and find out the maximum. 
+
+The illustration below shows how Beam Search can lead to a better sequence.
+![t = 1](https://i.imgur.com/biX7tAE.png)
+![t = 2](https://i.imgur.com/MdFAuV7.png)
+![t = 3](https://i.imgur.com/kuzYLq2.png)
```
Comment (Atcold) [inline docs/en/week11/11-2.md:36]: What is this???
Read the [instructions](https://atcold.github.io/pytorch-Deep-Learning/en/about/).

**Review by Atcold — CHANGES_REQUESTED** [review]
There are no figures, file name is wrong, math does not render.
Please, run the website locally to check that everything is displayed correctly.
[Here](https://github.com/Atcold/pytorch-Deep-Learning/blob/master/docs/README.md) are some instructions.
Also, read the [how to contribute](https://atcold.github.io/pytorch-Deep-Learning/en/about/) notes.
There are no summaries in the file `11.md`.
`11-2.md` seems way too short.

File: docs/en/week11/11-2.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+## Inference Time
+
+The inference of a transcription from a given audio signal can be formulated using 2 distributions:
+
+* The acoustic model (audio to transcription), represented as $P(Y|X)$
+* The language model, $P(Y)$
+
+The final inference is obtained by taking sum of the log probabilities of the above two, i.e.
+
+$$ Y* = \underset{Y}{\operatorname{argmax}} \log P(Y|X) + \log P(Y) $$
```
Comment (Atcold) [inline docs/en/week11/11-2.md:18]: Use `\mid` for the conditional bar, here and everywhere else.

File: docs/en/week11/11-2.md [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+We use the additional term to ensure the transcription is consistent with the rules of the language. Else we may get grammatically wrong transcription.
+
+
+## Beam Search
```
Comment (Atcold) [inline docs/en/week11/11-2.md:23]: Everything is still broken. Plz fix.

image

File: docs/en/week11/11-2.md [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+
+
+
+|  | NN | GTN |
+| -------- | -------- | -------- |
+| Core Data Structure     | Tensor     | Graph/WFSA     |
+|     | Matrix Multiplcation     | Compose     |
+| Core Operations  | Reduction operations (Sum, Product)| Shortest Distance (Viterbi,Forward) |
+| | Negate, Add, Subtract, ...     | Closure, Union, Concatenate, ...    |
+
```
Comment (Atcold) [inline docs/en/week11/11-2.md:56]: It seems rather short, this second part.
Are we sure nothing is missing?

**Review by Atcold — CHANGES_REQUESTED** [review]
Please, run the website locally and make sure everything actually displays correctly.

File: docs/en/week11/11-2.md (hunk 12 above)
Comment (Atcold) [reply docs/en/week11/11-2.md:23]: Plz, read the [instructions](https://atcold.github.io/pytorch-Deep-Learning/en/about/).

File: docs/en/week11/11-2.md (hunk 10 above)
Comment (gyaneshg96) [reply docs/en/week11/11-2.md:36]: I will change them

File: docs/en/week11/11-2.md (hunk 13 above)
Comment (gyaneshg96) [reply docs/en/week11/11-2.md:56]: Actually Abed  is yet to write his part for the remaining lecture. Hopefully he will get it done by a day or two.

Comment (Atcold) [reply docs/en/week11/11-2.md:56]: I see, I didn't know that, and I thought this was your complete submission.
Still, fix what's broken so far. Thaaaanks! 🙂

**Review by Atcold — CHANGES_REQUESTED** [review]
11-2 is completely broken.
**Do run the website locally before pushing anything.**

**Pull request merged**